core/src/main/java/tc/oc/pgm/stats/StatsModule.java

```diff
@@ -0,0 +1,56 @@
+package tc.oc.pgm.stats;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.logging.Logger;
+import net.kyori.adventure.text.Component;
+import org.bukkit.Color;
+import org.jdom2.Document;
+import org.jdom2.Element;
+import org.jetbrains.annotations.Nullable;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.math.Formula;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLFluentParser;
+import tc.oc.pgm.util.xml.XMLUtils;
+
+public class StatsModule implements MapModule<StatsMatchModule> {
+  private final List<StatType.OfFormula> formulaStats;
+
+  private StatsModule(List<StatType.OfFormula> formulaStats) {
+    this.formulaStats = formulaStats;
+  }
+
+  @Override
+  public @Nullable StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    return new StatsMatchModule(match, formulaStats);
+  }
+
+  public static class Factory implements MapModuleFactory<StatsModule> {
+    @Override
+    public StatsModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      XMLFluentParser parser = factory.getParser();
+      List<Element> statElements = XMLUtils.flattenElements(doc.getRootElement(), "stats");
+      if (statElements.isEmpty()) {
+        return new StatsModule(Collections.emptyList());
+      } else {
+        List<StatType.OfFormula> formulaStats = new ArrayList<>();
+        for (Element el : statElements) {
+          Component name = parser.component(el, "name").required();
+          Color color = parser.hexColor(el, "color").optional(Color.GREEN);
```

Comment:
As for parsing, everything else in pgm just uses `XMLUtils#parseChatColor` and then feeds it thru `TextFormatter#convert` at some point after parsing, that makes sure support for things like `aqua` etc still work, which parsing as hex color won't do.

Unless you're wanting to write a new `TextColor` parser that can take in both (which would be appreciated for future arbitrary rgb color support down the line!), you should stick to those mentioned above methods (but, feel free to wrap them in the xml fluent parser for future expansion for rgb support)

Reply:
I'll probably stick to the first method you've mentioned for this PR and wrap it in to the fluent parser to keep this PR relatively small